samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -2853,7 +2853,7 @@ public boolean getHasGradingData(Long publishedAssessmentId) {
    * @return a list of matching key values OR empty if none are found
    */
   private List<String> extractCalculatedQuestionKeyFromItemText(String itemText, Pattern identifierPattern) {
-      LinkedHashSet<String> keys = new LinkedHashSet<>();
+      List<String> keys = new ArrayList<>();
```

Comment:
We do a list with repeated values instead of a list with unique values.

Reply:
nice find @josecebe